core/state/statedb.go

```diff
@@ -374,14 +388,95 @@ func (s *StateDB) MergeSlotDB(slotDb *StateDB, slotReceipt *types.Receipt, txInd
 		if _, exist := s.stateObjectsDirty[addr]; !exist {
 			s.stateObjectsDirty[addr] = struct{}{}
 		}
-
+		// system address is EOA account, it should have no storage change
 		if addr == systemAddress {
 			continue
 		}
 
 		// stateObjects: KV, balance, nonce...
-		if obj, ok := slotDb.getStateObjectFromStateObjects(addr); ok {
-			s.storeStateObjectToStateDB(addr, obj.deepCopyForSlot(s))
+		dirtyObj, ok := slotDb.getStateObjectFromStateObjects(addr)
+		if !ok {
+			panic(fmt.Sprintf("MergeSlotDB dirty object not exist! (txIndex: %d, addr: %s)", slotDb.txIndex, addr.String()))
+		}
+		mainObj, exist := s.loadStateObjectFromStateDB(addr)
+
+		log.Debug("MergeSlotDB", "txIndex", slotDb.txIndex, "addr", addr,
+			"exist", exist, "dirtyObj.deleted", dirtyObj.deleted)
+		if !exist {
+			// addr not exist on main DB, do ownership transfer
+			dirtyObj.db = s
+			dirtyObj.finalise(true) // prefetch on dispatcher
+			s.storeStateObjectToStateDB(addr, dirtyObj)
+			delete(slotDb.parallel.dirtiedStateObjectsInSlot, addr) // transfer ownership
+		} else {
+			// addr already in main DB, do merge: balance, KV, code, State(create, suicide)
+			// can not do copy or ownership transfer directly, since dirtyObj could have outdated
+			// data(may be update within the conflict window)
+
+			// This is debug log for add balance with 0, can be removed
+			if dirtyObj.deleted && dirtyObj.empty() && !mainObj.empty() {
```

Comment:
can we delete the debug logs

Reply:
accept, this did not happen and should not happen.